### PR TITLE
Add disabled and label to button factory

### LINKED ISSUE #284: Unable to mark button as "disabled"
It seems not possible to mark a button as disabled in the steps provided, which is possible in the underlying Shepherd library.

### PR DESCRIPTION
**Pull request by rwwagner90**
Closes #284

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,60 +25,63 @@
...
-    "eslint-plugin-react": "^7.14.2",
-    "eslint-plugin-standard": "^4.0.0",
-    "gh-pages": "^2.0.1",
+    "eslint-plugin-react": "^7.19.0",
+    "eslint-plugin-standard": "^4.0.1",
+    "gh-pages": "^2.2.0",
     "install-peers-cli": "^2.2.0",
-    "react-scripts": "^3.4.0",
+    "jest-environment-jsdom-sixteen": "^1.0.3",
+    "react": "^16.8.0",
```
Comment (chuckcarpenter) [inline package.json:73]: These need to be peer dependencies and I have the hook for installing peers already for local dev.

File: src/index.tsx [hunk 2]
```diff
@@ -36,12 +36,14 @@ const addSteps = (steps: Array<Step.StepOptions>, tour: Tour) => {
 
     if (buttons) {
       step.buttons = buttons.map((button: ShepherdButtonWithType) => {
-        const { type, classes, text, action } = button;
+        const { action, classes, disabled, label, text, type } = button;
```
Comment (chuckcarpenter) [inline src/index.tsx:39]: I think updating dependencies should be for another PR, not this one.

File: package.json (hunk 1 above)
Comment (rwwagner90) [reply package.json:73]: They are peerDeps. They can be both peer and devDeps. Any reason we can't have them be both?

File: src/index.tsx (hunk 2 above)
Comment (rwwagner90) [reply src/index.tsx:39]: Totally get that, but wasn't sure what was causing everything to fail all the time. Not sure it hurts anything to update them.

File: package.json (hunk 1 above)
Comment (chuckcarpenter) [reply package.json:73]: but, they aren't dev deps? Not sure why it's needed to add here.
